packages/matter.js/src/node/Node.ts

```diff
@@ -129,9 +170,17 @@ export class Node<T extends RootEndpoint = RootEndpoint> extends Endpoint<T> {
         logger.notice(Diagnostic.strong(this.toString()), message);
     }
 
-    protected endpointCrashed(endpoint: Endpoint) {
-        return this.lifecycle.endpointError.emit(endpoint, endpoint.construction.error ?? new Error("Unknown error"));
+    override async [Construction.destruct]() {
+        await this.cancel();
```

Comment:
You mean this.close() instead of this.cancel()?  this.close() is what triggers `[Construction.destruct]`.  I added to the comment on `construction.close` to try to justify its purpose and use.

It would be more verbose but easier to understand if we got rid of `construction.close` and `[Construction.destruct]` and instead just do everything manually in `close`.  I'd be OK either way...  Figured we could try it this way and tear it out if it's annoying.

Reply:
Alright. Was just about that this class assumes that the parent does something in destruct (aka executing close). But fine because we still control what is done where.

Reply:
Oh, yes you couldn't invoke super if it wasn't implemented. This is a case of "I know my base supports destruction and I want to perform final teardown before it does". Doing it in [destruct] makes it so you don't need to await construction or handle exceptions yourself.